src/plugins/imagery/components/ImageryView.vue

```diff
@@ -1163,6 +1169,17 @@ export default {
             let isVisible = this.layers[index].visible === true;
             this.layers[index].visible = !isVisible;
             this.visibleLayers = this.layers.filter(layer => layer.visible);
+        },
+        thumbnailClassList(index) {
```

Comment:
This is not a good idea - I think a better approach is to either set the classList on mount or on change rather than call the method on every redraw.

Reply:
You could add a property called `thumbsClass` to the imageHistory items and reference them in the view later: https://github.com/nasa/openmct/blob/mct5327/src/plugins/imagery/mixins/imageryData.js#L147

Something like:
```
const thumbsClass = this.thumbsClassList(datum);

return {
                ...datum,
                formattedTime,
                url,
                time,
                imageDownloadName,
                thumbsClass
            };
```

Reply:
Shefali and I sync'd on this performance improvement and I've elected to keep the current implementation. The complexity cost of maintaining/looking up items from a pre-computed thumbnail classlist might negate the performance benefits of the relatively cheap logic. In testing it locally each comparison takes between 0 - 0.003 ms per thumbnail and would remain relatively cheap even if recomputed superfluously on rerender.